File: ARRAY/ARRAY/Linear_Search/LinearSearchArray.js
function search(arr, x) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === x) {
            return i; // Return if element is found
        }
    }
    return -1; // Return -1 if element is not found
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('Enter the size of the array: ', n => {
    n = parseInt(n);
    let arr = [];

    readline.question(`Enter ${n} elements:\n`, elements => {
        arr = elements.split(' ').map(Number);
        
        readline.question('Enter the target element to Search: ', x => {
            x = parseInt(x);
            let result = search(arr, x);
            if (result === -1) {
                console.log('Element is not present in array');
            } else {
                console.log(`Element found at index ${result}`);
            }
            readline.close();
        });
    });
});

File: ARRAY/ARRAY/Linear_Search/LinearSearchArray.cpp
#include <iostream>
using namespace std;

class LinearSearchArray {
public:
    int search(int arr[], int N, int x) {
        for (int i = 0; i < N; i++) {
            if (arr[i] == x)
                return i; // Return if element is found
        }
        return -1; // Return -1 if element is not found
    }
};

int main() {
    int N, x;
    cout << "Enter the size of the array: ";
    cin >> N;
    int arr[N];
    cout << "Enter " << N << " elements: " << endl;
    for (int i = 0; i < N; i++) {
        cin >> arr[i];
    }
    cout << "Enter the target element to search: ";
    cin >> x;
    
    LinearSearchArray ob;
    int result = ob.search(arr, N, x);
    
    if (result == -1) {
        cout << "Element is not present in array" << endl;
    } else {
        cout << "Element found at index " << result << endl;
    }
    
    return 0;
}